dask/dataframe/io/tests/test_parquet.py

```diff
@@ -4807,3 +4825,21 @@ def test_read_parquet_convert_string_fastparquet_warns(tmp_path):
             UserWarning, match="`dataframe.convert_string` is not supported"
         ):
             dd.read_parquet(outfile, engine="fastparquet")
+
+
+@PYARROW_MARK
+@pytest.mark.skipif(
+    not PANDAS_GT_200, reason="pd.Index does not support int32 before 2.0"
+)
+def test_read_parquet_preserve_categorical_column_dtype(tmp_path):
+    df = pd.DataFrame({"a": [1, 2], "b": ["x", "y"]})
+
+    outdir = tmp_path / "out.parquet"
+    df.to_parquet(outdir, engine="pyarrow", partition_cols=["a"])
+    ddf = dd.read_parquet(outdir, engine="pyarrow")
+
+    expected = pd.DataFrame(
+        {"b": ["x", "y"], "a": pd.Categorical(pd.Index([1, 2], dtype="int32"))},
+        index=[0, 0],
+    )
```

Comment:
Not meant as a blocking comment. FWIW I tend to prefer the approach you took in `test_partitioned_preserve_index` where we use the original `pandas` DataFrame instead of creating an entire new one. That makes it less cumbersome if, in the future, someone wants to update the data in the DataFrame (currently it would need to be changed in two places)